packages/ui5-application/templates/core/webapp/index.html

```diff
@@ -12,7 +12,10 @@
     </style>
     <script
         id="sap-ui-bootstrap"
-        src="resources/sap-ui-core.js"
+        src="<%= ui5.bootstrapSrc === 'Content delivery network (OpenUI5)' ?
+            'https://openui5.hana.ondemand.com/resources/sap-ui-core.js' :
+                ui5.bootstrapSrc === 'Content delivery network (SAPUI5)' ?
+                    'https://sapui5.hana.ondemand.com/resources/sap-ui-core.js' : 'resources/sap-ui-core.js'  %>"
```

Comment:
This approach is not really compatible with Fiori tools. 
We put the host the ui5*.yaml and proxy with https://www.npmjs.com/package/@sap/ux-ui5-tooling e.g.  cdn host is https://ui5.sap.com. This allows switching ui5 versions with the ui5*.yaml or launch config for preview
```
  customMiddleware:
    - name: fiori-tools-proxy
      afterMiddleware: compression
      configuration:
        ignoreCertError: false
        backend:
          - path: /sap
            url: https://somehost:44300
            client: "500"
            destination: ECXCLNT500
        ui5:
          path:
            - /resources
            - /test-resources
          url: https://ui5.sap.com
          version: ""
```
CDN can be passed into export const `getFioriToolsProxyMiddlewareConfig` to use a value other on than the default.
Can this approach work for easy-ui5 also?

Reply:
> This approach is not really compatible with Fiori tools.

given the type def...
```
export enum BootstrapSrc {
	CdnOpenUI5 = "Content delivery network (OpenUI5)",
	CdnSAPUI5 = "Content delivery network (SAPUI5)",
	Local = "Local Resources"
}
```
..., doing `ui5.bootStrapSrc = BootstrapSrc.Local` or simply leaving it `undefined`, this would result in 
```js
<script
        id="sap-ui-bootstrap"
        src="resources/sap-ui-core.js
// ...
>
```
which should be fine with the fiori tools proxy, right?

btw: the fiori tools proxy to the ui5 cdn has a problem in its' default definition anyway: given the default config you mentioned above...
```
  customMiddleware:
    - name: fiori-tools-proxy
      afterMiddleware: compression
      configuration:
        ignoreCertError: false
        backend:
          - path: /sap
            url: https://somehost:44300
            client: "500"
            destination: ECXCLNT500
        ui5:
          path:
            - /resources <-- this is a problem
            - /test-resources
          url: https://ui5.sap.com
          version: ""
```
...proxying `/resources` to the cdn prevents any ui5 tooling module shims from working. 
the shims need `resources/path/to/shim` working for the ui5 tooling.
so a solution could be to only proxy `sap` resources to the cdn:
```
 ui5:
          path:
            - /resources/sap*
            - /test-resources
```

Reply:
@vobu do you have an example project with a module shim? I am wondering if this is easier solvable by running the middleware after a different middleware i.e. after the module shims are resolved.

Otherwise, the change to `resources/sap*` looks reasonable as well but requires some testing on our side first.

Reply:
there's a whole thread with examples here: https://answers.sap.com/questions/13461279/ui5-project-shims-not-working.html?childToView=13479438